packages/viewer/src/services.ts

```diff
@@ -20,23 +20,19 @@ export class StatusError extends Error {
  * Helper function to process responses from the editor services.
  * These usually come wrapped in a JSON with a status field that can be either "ok" or "warning".
  * If status is "ok", return the result value along it. Otherwise, throw a StatusError.
- * @param {Promise<Response>} response - The response given by the service.
+ * @param {Response} response - The response given by the service.
  * @returns {Promise<any>} The unwrapped result of the response, if valid.
  * @throws {StatusError} Service responded with a non-ok status.
+ *                       If the response.json() fails, it'll return a generic exception.
  */
-export async function processJsonResponse(response: Promise<Response>): Promise<any> {
-  try {
-    const { status, result } = await (await response).json();
-
-    if (status !== "ok") {
-      throw new StatusError("Service responded with a non-ok status");
-    }
+async function processJsonResponse(response: Response): Promise<any> {
+  const { status, result } = await response.json();
 
-    return result;
-  } catch (e) {
-    // TODO manage network and status non-ok errors
-    throw e;
+  if (status !== "ok") {
```

Comment:
This assumes that the caller will allways provide a valid response, but that is not the case. For example, in the function `createImage`: 

```typescript
  // this promise could be rejected
  const response = await callService(params, "showimage", MethodType.Post, url, extension);
  return processJsonResponse(response);
```

In this case, the rejected promise is handled by the parent caller `renderMathML` in a try/catch block, resulting in a `invalid MathML format` error message, which is confusing if the mathml is perfectly fine and the call fails due to a service error.

Reply:
solved in the new version!